docs/src/main/java/org/geotools/tutorial/coverage/ImageTiler.java

```diff
@@ -0,0 +1,224 @@
+            System.exit(1);
+        }
+
+        tiler.tile();
+    }
+
+    private static void printUsage() {
+        System.out.println("Usage: -f inputFile -o outputDirectory [-tw tileWidth<default:256> "
+                + "-th tileHeight<default:256> ");
+        System.out.println("-htc horizontalTileCount<default:16> -vtc verticalTileCount<default:8>");
+    }
+    //docs end main
+
+    //docs start load coverage
+    private void tile() throws IOException {
+        AbstractGridFormat format = GridFormatFinder.findFormat(this.getInputFile());
+        String fileExtension = this.getFileExtension(this.getInputFile());
+
+        //working around a bug/quirk in geotiff loading via format.getReader which doesn't set this
+        //correctly
+        Hints hints = null;
+        if (format instanceof GeoTiffFormat) {
+            hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER,Boolean.TRUE);
+        }
+
+        GridCoverage2DReader gridReader = format.getReader(
+                this.getInputFile(),
+                hints);
+        GridCoverage2D gridCoverage = gridReader.read(null);
+        //docs end load coverage
+
+        //docs start envelope
+        Envelope2D coverageEnvelope = gridCoverage.getEnvelope2D();
+        double coverageMinX = coverageEnvelope.getBounds().getMinX();
+        double coverageMaxX = coverageEnvelope.getBounds().getMaxX();
+        double coverageMinY = coverageEnvelope.getBounds().getMinY();
+        double coverageMaxY = coverageEnvelope.getBounds().getMaxY();
+
+        int htc = this.getNumberOfHorizontalTiles() != null
+                ? this.getNumberOfHorizontalTiles() : NUM_HORIZONTAL_TILES;
+        int vtc = this.getNumberOfVerticalTiles() != null
+                ? this.getNumberOfVerticalTiles() : NUM_VERTICAL_TILES;
+
+        double geographicTileWidth = (coverageMaxX - coverageMinX) / htc;
+        double geographicTileHeight = (coverageMaxY - coverageMinY) / vtc;
+
+        CoordinateReferenceSystem targetCRS = gridCoverage.getCoordinateReferenceSystem();
+        //docs end envelope
+
+        //make sure to create our output directory if it doesn't already exist
+        File tileDirectory = this.getOutputDirectory();
+        if (!tileDirectory.exists()) {
+            tileDirectory.mkdirs();
+        }
+
+        //iterate over our tile counts
+        for (int i = 0; i < htc; i++) {
+            for (int j = 0; j < vtc; j++) {
+
+                //create the envelope of the tile
+                double envelopeStartX = (i * geographicTileWidth) + coverageMinX;
+                double envelopeEndX = envelopeStartX + geographicTileWidth;
+                double envelopeStartY = (j * geographicTileHeight) + coverageMinY;
+                double envelopeEndY = envelopeStartY + geographicTileHeight;
+
+                Envelope envelope = new ReferencedEnvelope(
+                        envelopeStartX, envelopeEndX, envelopeStartY, envelopeEndY, targetCRS);
```

Comment:
Can we factor this out into a separate method for readability :)

It will make the code easier to follow, and will be easier to present as a "step" in the tutorial - and we get to have javadocs.